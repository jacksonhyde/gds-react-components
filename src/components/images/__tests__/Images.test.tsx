import React from 'react';
import { shallow } from 'enzyme';
import Images from '..';

describe('Images', () => {
  it('matches snapshot', () => {
    const component = shallow(<Images />);
    expect(component.hasClass('govuk-image')).toBeTruthy();
    expect(component).toMatchSnapshot('Images');
    component.unmount();
  });

  it('renders caption', () => {
    const component = shallow(<Images caption="Caption" />);
    expect(component.find('figcaption').exists()).toBeTruthy();
    expect(component.find('figcaption').hasClass('govuk-image__caption')).toBeTruthy();
    expect(component.find('figcaption').text()).toBe('Caption');
    component.unmount();
  });
});
