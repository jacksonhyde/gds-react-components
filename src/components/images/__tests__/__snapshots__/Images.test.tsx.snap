// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images matches snapshot: Images 1`] = `
<figure
  className="govuk-image"
>
  <img
    className="govuk-image__img"
  />
</figure>
`;
